### DISCUSSION TITLE
BLE fails on STM32WB55 Nucleo board

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#16177] [STM32 / Pyboard] [answered]
I managed once (1 year ago) to get the aioble example to work on a stm32wb55 board. I now got a new STM32WB55 Nucleo board and I get `tl_ble_wait_resp: timeout.` and no other debug info. It has somethin to do with the RFcore, but I have no clue where to start...

There are some rfcore related files in the repo... Can anyone explain what I have to do with the following files in the Micropython [repo directory of the STM32WB55 Nucleo](https://github.com/micropython/micropython/tree/master/ports/stm32/boards/NUCLEO_WB55):
* rfcore_debug.py
* rfcore_firmware.py
* rfcore_makefirmware.py

Thanks!

**Comment by andrewleech** [comment]
It's been reported that the latest WS radio firmware from ST breaks compatibility with micropython, for details see https://github.com/orgs/micropython/discussions/15522

The `rfcore_makefirmware.py` script is intended to convert the radio firmware files provided by ST into a format that can be loaded onto the WB55 while running micropython, after which `rfcore_firmware.py` can be used to flash the radio firmware (FUS & WS).

**Reply by saraverbeecke** [reply]
Does the `rfcore_makefirmware.py` work when using the older 1.13 version from ST?

**Reply by andrewleech** [reply, marked as answer]
Yes it should, that's likely the last version it was really tested with

**Reply by saraverbeecke** [reply]
I just used the v1.13 release and I can confirm it works. Thanks!

**Comment by andrewleech** [comment]
I haven't looked into it any deeper yet however on my NUCLEO_WB55 I just installed the v1.21.0 `stm32wb5x_BLE_HCILayer_extended_fw.bin` stack from:
* https://github.com/STMicroelectronics/STM32CubeWB/blob/v1.21.0/Projects/STM32WB_Copro_Wireless_Binaries/STM32WB5x/stm32wb5x_BLE_HCILayer_extended_fw.bin

It was loaded at address 0x080DA000 for my STM32WB5xxG (1MB Flash) chip as per the [release note](https://htmlpreview.github.io/?https://github.com/STMicroelectronics/STM32CubeWB/blob/v1.21.0/Projects/STM32WB_Copro_Wireless_Binaries/STM32WB5x/Release_Notes.html).
The basic ble active call worked correctly without timeout
```
>>> from bluetooth import  BLE
>>> BLE().active(True)
True
```
